src/NeoSharp.Core/Persistence/IDbModel.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Threading.Tasks;
+using NeoSharp.Core.Types;
+
+namespace NeoSharp.Core.Persistence
+{
+    public interface IDbModel
```

Comment:
IDbModel<TEntity> ?

Reply:
It's on the model that I post on discord. 
The entry point on each database implementation will be the DbModel and internal will be the DbContext

With this the application interact with the database implementation using the <TEntity> and each implementation know if going to save direct the object, will serialize like this or that.
The point of communication don't need to know anything about the real DB implemantation